app/src/main/java/de/danoeh/antennapod/activity/WidgetConfigActivity.java

```diff
@@ -0,0 +1,88 @@
+
+    private static final String WIDGET_PREFS = "widget_preferences";
+    private static final String WIDGET_COLOR = "widget_color";
+
+    private int appWidgetId = AppWidgetManager.INVALID_APPWIDGET_ID;
+
+    private SeekBar opacitySeekBar;
+    private TextView opacityTextView;
+    private RelativeLayout widgetPreview;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setTheme(UserPreferences.getTheme());
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_widget_config);
+
+        Intent configIntent = getIntent();
+        Bundle extras = configIntent.getExtras();
+        if (extras != null) {
+            appWidgetId = extras.getInt(AppWidgetManager.EXTRA_APPWIDGET_ID,
+                    AppWidgetManager.INVALID_APPWIDGET_ID);
+        }
+
+        Intent resultValue = new Intent();
+        resultValue.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,appWidgetId);
+        setResult(RESULT_CANCELED,resultValue);
+        if (appWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+            finish();
+        }
+
+        opacityTextView = findViewById(R.id.widget_opacity_textView);
+        opacitySeekBar = findViewById(R.id.widget_opacity_seekBar);
+        widgetPreview = findViewById(R.id.widgetLayout);
+        opacitySeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+            @Override
+            public void onProgressChanged(SeekBar seekBar, int i, boolean b) {
+                opacityTextView.setText(seekBar.getProgress() + "%");
+                widgetPreview.setBackgroundColor(getAlphaFromPercentage(opacitySeekBar.getProgress()));
+            }
+            @Override
+            public void onStartTrackingTouch(SeekBar seekBar) {
+            }
+            @Override
+            public void onStopTrackingTouch(SeekBar seekBar) {
+            }
+        });
+    }
+
+    public void confirmCreateWidget(View v){
+        AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(this);
+
+        RemoteViews views = new RemoteViews(this.getPackageName(), R.layout.player_widget);
+        views.setInt(R.id.widgetLayout,"setBackgroundColor", getAlphaFromPercentage(opacitySeekBar.getProgress()));
+        appWidgetManager.updateAppWidget(appWidgetId,views);
+
+        SharedPreferences prefs = getSharedPreferences(WIDGET_PREFS, MODE_PRIVATE);
+        SharedPreferences.Editor editor = prefs.edit();
+        editor.putInt(WIDGET_COLOR + appWidgetId,getAlphaFromPercentage(opacitySeekBar.getProgress()));
+        editor.apply();
+
+        Intent resultValue = new Intent();
+        resultValue.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID,appWidgetId);
+        setResult(RESULT_OK,resultValue);
+        finish();
+    }
+
+    public int getAlphaFromPercentage(int opacityPercentage){
+        return 0x00262C31+0xFF000000*(255-(int)Math.round(opacityPercentage*2.54));
```

Comment:
Opacity of 0 means that it is fully transparent. The method does it the other way around, I think. Additionally, please add spaces around +-* and after (int). The method name is a bit misleading because it generates the whole color, not just alpha.

Reply:
I am not completely sure why but it did work like this. :D

When I use 0x00262C31+0x1000000*alpha ,then the color is transparent with alpha = 0 and opaque when alpha = 255 so maybe I should use this solution.

Reply:
```
percentage = seekBar.getProgress();
color = DEFAULT_COLOR + 0xFF000000 * (1 - 0.01 * percentage);
```

With something like this, you don't need to multiply with 2.55 (which looks a bit strange).

Reply:
I totally understand how this should work, but unfortunately somehow it do not. When I use this equation I end up with different colors with full opacity. Seems like rounding to integer keeps messing numbers up.

How about this? 
```
public int getColorWithAlpha(int color,int opacity) {
        return  (int)Math.round(0xFF * (0.01 * opacity)) << 24 | color ;
    }
```
Or is version without binary operators prefered ? 

```
public int getColorWithAlpha(int color,int opacity) {
        return  (int)Math.round(0xFF * (0.01 * opacity)) * 0x1000000 + color ;
    }
```

Reply:
Oooh, of course. You are right. Using the number directly could produce stuff like 0xff123456 which influences other colors.

Both versions in your comment look nice. Take the one you like more ;)